parsers/src/main/java/org/chocosolver/parser/dimacs/DIMACS.java

```diff
@@ -81,20 +83,44 @@ public void createSolver() {
     public void buildModel() {
         List<Model> models = portfolio.getModels();
         for (int i = 0; i < models.size(); i++) {
+            Model m = models.get(i);
+            Solver s = m.getSolver();
             try {
                 long ptime = -System.currentTimeMillis();
-                parse(models.get(i), parsers[i], i);
-                models.get(i).getSolver().logWithANSI(ansi);
+                parse(m, parsers[i], i);
+                if (!logFile.equals("")) {
+                    s.log().remove(System.out);
+                    s.log().add(new PrintStream(Files.newOutputStream(Paths.get(logFile)), true));
+                }else {
+                    s.logWithANSI(ansi);
+                }
                 if (level.isLoggable(Level.INFO)) {
-                    models.get(i).getSolver().log().white().printf("File parsed in %d ms%n", (ptime + System.currentTimeMillis()));
+                    s.log().white().printf("File parsed in %d ms%n", (ptime + System.currentTimeMillis()));
                 }
                 if (level.is(Level.JSON)) {
-                    models.get(i).getSolver().log().printf("{\"name\":\"%s\",\"stats\":[", instance);
+                    s.getMeasures().setReadingTimeCount(System.nanoTime() - s.getModel().getCreationTime());
+                    s.log().printf(Locale.US,
+                            "{\t\"name\":\"%s\",\n" +
+                                    "\t\"variables\": %d,\n" +
+                                    "\t\"constraints\": %d,\n" +
+                                    "\t\"policy\": \"%s\",\n" +
+                                    "\t\"parsing time\": %.3f,\n" +
+                                    "\t\"building time\": %.3f,\n" +
+                                    "\t\"memory\": %d,\n" +
+                                    "\t\"stats\":[",
+                            instance,
+                            m.getNbVars(),
+                            m.getNbCstrs(),
+                            m.getSolver().getObjectiveManager().getPolicy(),
+                            (ptime + System.currentTimeMillis()) / 1000f,
+                            s.getReadingTimeCount(),
+                            m.getEstimatedMemory() // TODO
```

Comment:
don't forget the TODO

] is missing

Reply:
The TODO is useless, I'll remove it.
The `]` is created on exit, so no need to write it here